pkg/cloud/cloud_test.go

```diff
@@ -2509,6 +2509,127 @@ func TestResizeOrModifyDisk(t *testing.T) {
 	}
 }
 
+func TestModifyTags(t *testing.T) {
+
+	validTagsToAddInput := map[string]string{
+		"key1": "value1",
+		"key2": "value2",
+		"key3": "",
+	}
+
+	validTagsToDeleteInput := []string{
+		"key1",
+		"key2",
+	}
+
+	emptyTagsToAddInput := map[string]string{}
+	emptyTagsToDeleteInput := []string{}
+
+	testCases := []struct {
+		name              string
+		volumeID          string
+		createTags        bool
+		deleteTags        bool
+		negativeCase      bool
+		modifyDiskOptions *ModifyDiskOptions
+		ModifyTagsOptions *ModifyTagsOptions
+		expErr            error
+		mockEC2           *MockEC2API
+		volumes           []types.Volume
+	}{
+		{
+			name:       "success normal tag addition",
+			volumeID:   "mod-tag-test-name",
+			createTags: true,
+			ModifyTagsOptions: &ModifyTagsOptions{
+				TagsToAdd: validTagsToAddInput,
+			},
+			expErr: nil,
+		},
+		{
+			name:       "success normal tag deletion",
+			volumeID:   "mod-tag-test-name",
+			deleteTags: true,
+			ModifyTagsOptions: &ModifyTagsOptions{
+				TagsToDelete: validTagsToDeleteInput,
+			},
+			expErr: nil,
+		},
+		{
+			name:       "success normal tag addition and tag deletion",
+			volumeID:   "mod-tag-test-name",
+			createTags: true,
+			deleteTags: true,
+			ModifyTagsOptions: &ModifyTagsOptions{
+				TagsToAdd:    validTagsToAddInput,
+				TagsToDelete: validTagsToDeleteInput,
+			},
+			expErr: nil,
+		},
+		{
+			name:         "fail: EC2 API generic error TagsToAdd",
+			createTags:   true,
+			negativeCase: true,
+			expErr:       fmt.Errorf("Generic EC2 API error"),
+			ModifyTagsOptions: &ModifyTagsOptions{
+				TagsToAdd:    validTagsToAddInput,
+				TagsToDelete: emptyTagsToDeleteInput,
+			},
+		},
+		{
+			name:         "fail: EC2 API generic error TagsToDelete",
+			deleteTags:   true,
+			negativeCase: true,
+			expErr:       fmt.Errorf("Generic EC2 API error"),
+			ModifyTagsOptions: &ModifyTagsOptions{
+				TagsToAdd:    emptyTagsToAddInput,
+				TagsToDelete: validTagsToDeleteInput,
+			},
+		},
+	}
+	for _, tc := range testCases {
```

Comment:
np: add an empty line between the end of the test cases and the for loop to match the style of other test functions in this file